consul/client.go

```diff
@@ -12,14 +12,51 @@ import (
 	"time"
 
 	"github.com/hashicorp/consul/consul/structs"
+	"github.com/hashicorp/consul/lib"
 	"github.com/hashicorp/serf/coordinate"
 	"github.com/hashicorp/serf/serf"
 )
 
 const (
```

Comment:
Agreed.  If possible I want to defer pushing this into the config struct in order to make a second, more mechanical pass that exposes many different const parameters as tunables to users.  I'm making the assumption that all tunables will require testing with fuzzed values and various runtime checks to make sure preconditions are met, and this will be one of many such places where that will need to happen.